src/commcare_cloud/terraform/modules/iam/EnforceMFAPolicy.json

```diff
@@ -0,0 +1,98 @@
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "AllowAllUsersToListAccounts",
+            "Effect": "Allow",
+            "Action": [
+                "iam:ListAccountAliases",
+                "iam:ListUsers",
+                "iam:ListVirtualMFADevices",
+                "iam:GetAccountPasswordPolicy",
+                "iam:GetAccountSummary"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "AllowIndividualUserToSeeAndManageOnlyTheirOwnAccountInformation",
+            "Effect": "Allow",
+            "Action": [
+                "iam:ChangePassword",
+                "iam:CreateAccessKey",
+                "iam:CreateLoginProfile",
+                "iam:DeleteAccessKey",
+                "iam:DeleteLoginProfile",
+                "iam:GetLoginProfile",
+                "iam:ListAccessKeys",
+                "iam:UpdateAccessKey",
+                "iam:UpdateLoginProfile",
+                "iam:ListSigningCertificates",
+                "iam:DeleteSigningCertificate",
+                "iam:UpdateSigningCertificate",
+                "iam:UploadSigningCertificate",
+                "iam:ListSSHPublicKeys",
+                "iam:GetSSHPublicKey",
+                "iam:DeleteSSHPublicKey",
+                "iam:UpdateSSHPublicKey",
+                "iam:UploadSSHPublicKey"
+            ],
+            "Resource": "arn:aws:iam::*:user/${aws:username}"
+        },
+        {
+            "Sid": "AllowIndividualUserToViewAndManageTheirOwnMFA",
+            "Effect": "Allow",
+            "Action": [
+                "iam:CreateVirtualMFADevice",
+                "iam:DeleteVirtualMFADevice",
+                "iam:EnableMFADevice",
+                "iam:ListMFADevices",
+                "iam:ResyncMFADevice"
+            ],
+            "Resource": [
+                "arn:aws:iam::*:mfa/${aws:username}",
+                "arn:aws:iam::*:user/${aws:username}"
+            ]
+        },
+        {
+            "Sid": "AllowIndividualUserToDeactivateOnlyTheirOwnMFAOnlyWhenUsingMFA",
```

Comment:
Does this mean anyone can disable MFA for their own account?

Reply:
Yeah it does—but when they disable their MFA device, they'll no longer be able to enter heightened privilege mode. Without an MFA device configured—and a session created based on a token from that device—the user can only do very few things, such as look at the list of users and change their own password and MFA device.

Reply:
This is I think a bigger gap in my mind actually—if a someone malicious gained access to a user's access keys (but didn't have access to their MFA device i.e. phone), they could still use that to log in and change the MFA device to be their own phone, and then log in with that. I haven't tried that escalation attack, but I think it's possible, and straightforward

Reply:
The second one is pretty critical since it means the limit is easily disabled if someone malicious gains access. However, using MFA still seems better than not using it since it means local auth tokens stored on disk have an expiration date.

Reply:
Okay turns out that was a false alarm—it was something that occurred to me as a potential vulnerability that I hadn't actually vetted yet. It turns out, and this makes sense, that the people who put together the policy suggestion and the docs to promote it https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_users-self-manage-mfa-and-creds.html thought this through. I tried the attack both through the console and through the command line. From the console, I can't complete login without MFA anyway (once I've set a device), so that's the end of the game. (Going to other pages directly redirects to sign in.) From the command line, when I send a command to enable a new MFA device for my user, it says it can't because I'd have too many devices. If I try to deactivate my current one to make room for the new one, it says I don't have permissions to deactivate. Basically it does exactly the right things in these scenarios.

Sorry for the false alarm!